File: Two Sum/Two Sum_hash.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> tmp;
        for(int i = 0; i < nums.size(); i++){
            if(tmp.find(target - nums[i]) != tmp.end()){
                return {tmp[target - nums[i]], i};
            }
            else{
                tmp[nums[i]] = i;
            }
        }
        return {};
    }
};

File: Two Sum/Two Sum_hash.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> res = new HashMap<Integer, Integer>();
        int[] ress = new int[2];
        for(int i = 0; i < nums.length; i++){
            if(res.containsKey(nums[i])){
                ress[0] = res.get(nums[i]);
                ress[1] = i;
            }else{
                res.put(target-nums[i], i);
            }
        }
        return ress;
    }
}